File: ce/customerengagement/on-premises/snippets/visualbasic/CRMV8/queryvb/vb/linqexamples40.vb
' <snippetlinqexamples40>


 Using orgSvcContext As New OrganizationServiceContext(_serviceProxy)
  Dim query_dejoin = From c In orgSvcContext.CreateQuery("contact") _
                     Join a In orgSvcContext.CreateQuery("account")
                     On c("contactid") Equals a("primarycontactid") _
                     Join l In orgSvcContext.CreateQuery("lead") _
                     On a("originatingleadid") Equals l("leadid") _
                     Select New With {Key .Contact = c,
                                      Key .Account = a,
                                      Key .Lead = l}
  For Each c In query_dejoin
   Console.WriteLine(c.Account.Attributes("name").ToString() _
                     &amp; " " _
                     &amp; c.Contact.Attributes("fullname").ToString() _
                     &amp; " " _
                     &amp; c.Lead.Attributes("leadid").ToString())
  Next c
 End Using

' </snippetlinqexamples40>

File: ce/customerengagement/on-premises/snippets/csharp/CRMV8/query/cs/linqexamples40.cs
// <snippetlinqexamples40>


 using (OrganizationServiceContext orgSvcContext = new OrganizationServiceContext(_serviceProxy))
 {
  var query_dejoin = from c in orgSvcContext.CreateQuery("contact")
                     join a in orgSvcContext.CreateQuery("account") 
                     on c["contactid"] equals a["primarycontactid"]
                     join l in orgSvcContext.CreateQuery("lead") 
                     on a["originatingleadid"] equals l["leadid"]
                     where (string)c["lastname"] != "Parker"
                     select new { Contact = c, Account = a, Lead = l };
  foreach (var c in query_dejoin)
  {
   System.Console.WriteLine(c.Account.Attributes["name"] + " " + 
    c.Contact.Attributes["fullname"] + " " + c.Lead.Attributes["leadid"]);
  }
 }

// </snippetlinqexamples40>